maintenance/pin.sh

```diff
@@ -55,33 +54,29 @@ PY_311_SCRIPTS=(
     .
 )
 
-RUNCMD="RUN apt-get update && \
-    apt-get install -y \
-        gir1.2-ostree-1.0 \
-        libgirepository1.0-dev \
-        && \
-    rm -rf /var/lib/apt/lists/* && \
-    pip install --upgrade pip && \
-    pip install pip-compile-multi
-"
-
-
+# Create a list of directories that need to be compiled for each python version
 PY38_DIRS=()
 PY311_DIRS=()
 for idx in "${!DIRS[@]}"; do
-    if [[ ${PY_38_SCRIPTS[@]} =~ "${DIRS[$idx]}" ]]; then
-        PY38_DIRS+=("${DIRS[$idx]}")
-    fi
-    if [[ ${PY_311_SCRIPTS[@]} =~ "${DIRS[$idx]}" ]]; then
-        PY311_DIRS+=("${DIRS[$idx]}")
-    fi
+    for script in "${PY_38_SCRIPTS[@]}"; do
+        if [[ "${DIRS[$idx]}" == "$script" ]]; then
+            PY38_DIRS+=("${DIRS[$idx]}")
+        fi
+    done
+    for script in "${PY_311_SCRIPTS[@]}"; do
+        if [[ "${DIRS[$idx]}" == "$script" ]]; then
+            PY311_DIRS+=("${DIRS[$idx]}")
+        fi
+    done
 done
 
 if [ ${#PY38_DIRS} -gt 0 ]; then
-    printf "FROM python:3.8\n${RUNCMD}" | docker build --platform linux/x86_64 --pull --tag "scriptworker-script-pin:3.8" -
-    echo "${PY38_DIRS[@]}" | xargs -n4 -P4 docker run --platform linux/x86_64 --rm -t -v "$PWD":/src -e EXTRA_ARGS="$EXTRA_ARGS" -e SUFFIX=py38.txt -w /src scriptworker-script-pin:3.8 maintenance/pin-helper.sh
+    # Make sure python 3.8 is installed
+    uv python install 3.8
```

Comment:
IMO we should still be doing this in `docker`, or finding some way to do this outside of it that doesn't affect local state. (It is surprising behaviour that running this script would affect what python versions I have available...)

Reply:
Even though `uv` uses standalone python distributions ([ref](https://docs.astral.sh/uv/guides/install-python/#getting-started)), I agree this script shouldn't be doing that for you.

I pondered for a bit and decided to just error out in case it can't find the required python versions. I leaned towards this instead of keep using docker.

Reply:
Can you please reconsider the docker part?

Reply:
@jcristau could you please elaborate?
`uv` handles different OS's with `--universal`, which in my mind was the reason why we used docker to begin with. Maybe I'm missing other benefits?

Reply:
Mostly that I have docker, I don't have uv, I'd rather not have yet another tool to remember to maintain/update, so this doesn't work for me right now.